enhancements/microshift/microshift-default-cni.md

```diff
@@ -0,0 +1,290 @@
+the target platforms for MicroShift are small SBCs or SoCs where most of
+the computing capability must remain available for the final applications,
+otherwise MicroShift stops being useful in it's target environment.
+
+### User Stories
+
+An application developer builds applications where:
+1. Pods need to talk to each other via TCP o UDP connectivity.
+2. Pods need to talk to services on the LAN or Internet via TCP or UDP.
+3. Some pods need be exposed as NodePort types of services on the LAN.
+4. Some pods need to be exposed to other pods as ClusterIP services types.
+
+An application developer builds applications where some pods must
+have limited connectivity to other pods, or to the external network (i.e.
+plugin services from vendors which must only contact specific internal APIs)
+
+The edge device with MicroShift and the final application is deployed on an
+IPv6, or DualStack IPV4/IPV6 network.
+
+As a device owner I can protect the edge device with a host-level firewall
+(firewalld, iptables, nftables), and that type of configuration is compatible
+with the CNI.
+
+The edge device should be capable of deploying on a network with dynamic address
+provisioning like DHCP, SLAAC, or DHCPv6 and remain functional when the IP address
+changes.
+
+### Goals
+ 
+* Implement a default CNI that satisfies the described user histories while
+  minimizing the impact on footprint.
+
+* The CNI plugin should run without the ClusterNetworkOperator, providing
+  the `microshift-networking` elements any basic system configuration elements,
+  tools or dependencies to make the CNI plugin work.
+
+* Follow the [MicroShift design
+  principles](https://github.com/openshift/microshift/blob/main/docs/design.md)
+
+### Non-Goals
+
+* We do not plan to support multi-node clusters.
+* We are not trying to address high availability within the CNI, 
+  high availability should be acomplished by having multiple edge devices working
+  as single node MicroShifts.
+
+* LoadBalancer type of Services, as this is out of the scope of the CNI.
+
+## Proposal
+
+To meet the requirements described in the user histories, as well footprint
+requirements of the edge environment, an optimized version of OVN Kubernetes
+is proposed.
+
+This option aligns better with Red Hat's strategy on networking and avoids
+the need to maintain bridge-cni which is deprecated in podman 4.
+
+Using OVN-kubernetes is only possible thanks to the following optimizations:
+* Single node operation
+* Communication via unix sockets, avoiding the need of TLS & Certificate creation
+  at boot time
+* No leader election, as there is no high availability or multi-node operation.
+* OpenvSwitch and ovsdb-server RAM optimization (CPUAffinity=0 and --no-mlockall)
+* Workload partitioning, to limit the cpusets where specific ovn-kubernetes workloads
+  can run.
+* New ovn-kubernetes with reduced footprint, oriented to single node operation.
```

Comment:
Maybe this can be expanded to:
`1)` new ovn-kubernetes image
`2)` optimized ovn-kubernetes deployment
`3)` managed by microshift main binary, instead of cluster network operator

`1)` contains improvement for reduced ovnk binaries and rpm packages.
`2)` contains improvement for ovnk manifests which remove containers such as ovn-dbchecker, kube-rbac-proxy etc and combine ovnk master&node to single container.